### PR TITLE
Update post

### LINKED ISSUE #10: Update Post
**Requirements**

- Once a post is updated, it’s marked as updated

### PR DESCRIPTION
**Pull request by CoBatool**
Closes #10 

- Let user update post
- Show post as updated when someone edits a post
- Update column in DB

**Missing:** 
- Edit attachment

**Edit button**

**When clicked on edit possibility to edit title/message**

**Once edited, show timestamp**

**Database updated**

### CODE REVIEW

**Review by mjimenez98 — APPROVED** [review]
This is pretty cool! Love it ❤️ 

One non-blocking thing I noticed is that if I click on edit and I don't change anything, the message will still pop up saying I edited the post. Would it be possible to verify if the original title or message has changed before displaying the edit timestamp?

Also, I would suggest running `Code -> Reformat Code` or `Alt + Command + L` on `PostServlet.java` and `DBPost.java` to get the formatting in line with main

File: src/main/webapp/Posts.jsp [hunk 1]
```diff
@@ -1,67 +1,80 @@
...
+        <input type="hidden" name="id" value="<%= post.getId() %>">
+        <input type="submit" name="request" class="btn button-color" value="edit">
+        <input type="submit" name="request" class="btn button-color" value="delete">
+    </form>
+    <% } %>
+    <hr/>
+    <%
+            }
+        }
+    %>
```
Comment (mjimenez98) [inline src/main/webapp/Posts.jsp:56]: I am finding it a bit hard to follow where things are being opened and closed, what's in where and so on 😕 . Would it be possible to give it some formatting or different indentation?

Comment (CoBatool) [reply src/main/webapp/Posts.jsp]: I just updated the file with `Ctrl+Alt+L` !   Is this better? Or do you want me to change it manually 😊?

**Comment by CoBatool** [comment]
> This is pretty cool! Love it ❤️
> 
> One non-blocking thing I noticed is that if I click on edit and I don't change anything, the message will still pop up saying I edited the post. Would it be possible to verify if the original title or message has changed before displaying the edit timestamp?
> 
> Also, I would suggest running `Code -> Reformat Code` or `Alt + Command + L` on `PostServlet.java` and `DBPost.java` to get the formatting in line with main



> This is pretty cool! Love it ❤️
> 
> One non-blocking thing I noticed is that if I click on edit and I don't change anything, the message will still pop up saying I edited the post. Would it be possible to verify if the original title or message has changed before displaying the edit timestamp?
> 
> Also, I would suggest running `Code -> Reformat Code` or `Alt + Command + L` on `PostServlet.java` and `DBPost.java` to get the formatting in line with main

Great!!! Thanks I changed it so the timestamp would be shown only if it is different than the original post.  Also fixed the formatting, thank you ! 🧐

File: src/main/webapp/Posts.jsp (hunk 1 above)
Comment (mjimenez98) [reply src/main/webapp/Posts.jsp]: _Manually_ 🥴

I'm sorry 😭  In [this excerpt](https://github.com/mjimenez98/message-board/pull/27/files#diff-9f7468293ceb1b36c8984e8cc2f57426f7556a4f554810d2767f595764859149R18-R24) for example the closing tag of the if is not aligned with the declaration.

But it's been approved so please don't feel like you have to do this. The front-end will be revamped soon.

**Comment by mjimenez98** [comment]
@CoBatool It is trivial but I just realized a potential experience issue: if the contents of the message are too long, it will be pretty hard to edit it since you will need to scroll right to fix, for example, a typo.

Would it work if the message input was changed to a textarea?

**Comment by CoBatool** [comment]
> @CoBatool It is trivial but I just realized a potential experience issue: if the contents of the message are too long, it will be pretty hard to edit it since you will need to scroll right to fix, for example, a typo.
> 
> Would it work if the message input was changed to a textarea?

AH, good point!!! Fixed ✨
